xenstoreclient/xenstore.go

```diff
@@ -529,7 +536,23 @@ func (xs *CachedXenStore) StopWatch() error {
 }
 
 func (xs *CachedXenStore) Clear() {
-	xs.writeCache = make(map[string]string, 0)
+	xs.writeCache = make(map[string]ValueUpdate, 0)
+}
+
+func (xs *CachedXenStore) UpdateCache() error {
+	for key, value := range xs.writeCache {
+		if value.update {
+			value.update = false
+			xs.writeCache[key] = value
+		} else {
+			delete(xs.writeCache, key)
+			err := xs.Rm(key)
+			if err != nil {
```

Comment:
Should be safer if we keep the cache if xs.Rm filed. It can be triggered by next refreash